controllers/awsmachine_controller.go

```diff
@@ -622,10 +651,40 @@ func (r *AWSMachineReconciler) deleteEncryptedBootstrapDataSecret(machineScope *
 	return nil
 }
 
-func (r *AWSMachineReconciler) createInstance(ec2svc services.EC2Interface, machineScope *scope.MachineScope, clusterScope cloud.ClusterScoper) (*infrav1.Instance, error) {
+func (r *AWSMachineReconciler) deleteBootstrapData(machineScope *scope.MachineScope, objectStoreSvc services.ObjectStoreInterface) error {
```

Comment:
What comment?

Reply:
For the section that is only needed for ignition based bootstrap.
Since ignition and cloudInit share several methods and has if/else logic, would be helpful to have a clear distinction with the comments to understand which code is for what bootstrap type.

Reply:
Hmm, I'm not sure where to put those comments and with what content really. Can you please provide some specific suggestions?

Generally speaking, language names (variables, functions, structs etc) should be preferred over comments as comments tend to get out of date, as they are not formally bound to the code.

This PR adds the following functions:
- `cloudInitUserData()`
- `ignitionUserData()`
- `deleteBootstrapData()` (paired with existing `resolveUserData()`)
- `deleteIgnitionBootstrapDataFromS3()`

In my opinion all of those functions follow very clear naming and it's clear from looking at the code which are ignition-specific.